python/ray/serve/_private/application_state.py

```diff
@@ -430,6 +434,26 @@ def is_deleted(self) -> bool:
         """
         return self._target_state.deleting and len(self._get_live_deployments()) == 0
 
+    def should_autoscale(self):
+        return (
+            self._target_state.config is not None
+            and self._target_state.config.autoscaling_policy is not None
+        )
+
+    def autoscale(self):
+        deployments: Dict[str, DeploymentDetails] = self.list_deployment_details()
+        decisions: Dict[str, int] = (
+            self._autoscaling_state_manager.apply_autoscaling_decision(deployments)
+        )
+
+        for deployment_name, decision_num_replicas in decisions.items():
+            deployment_id: DeploymentID = DeploymentID(
+                name=deployment_name, app_name=self._name
+            )
+            self._deployment_state_manager._deployment_states[deployment_id].autoscale(
+                target_num_replicas=decision_num_replicas
+            )
```

Comment:
i think we should create public functions on `self._deployment_state_manager` for the autoscaling purpose instead of accessing private attribute `_deployment_states`